core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -710,6 +710,26 @@ class LogCleanerTest extends JUnitSuite {
       assertEquals(offset, value)
     }
   }
+  
+  @Test
+  def testCleanTombstone(): Unit = {
+    val logConfig = LogConfig(new Properties())
+
+    val log = makeLog(config = logConfig)
+    val cleaner = makeCleaner(10)
+    
+    // Append a message with a large timestamp.
+    log.append(TestUtils.singleMessageSet(payload = "0".getBytes(),
+                                          key = "0".getBytes(),
+                                          timestamp = time.milliseconds() + logConfig.deleteRetentionMs + 10000))
+    log.roll()
+    cleaner.clean(LogToClean(TopicAndPartition("test", 0), log, 0, log.activeSegment.baseOffset))
+    // Append a tombstone with a small timestamp and roll out a new log segment.
+    log.append(TestUtils.singleMessageSet(payload = null, key = "0".getBytes(), timestamp = time.milliseconds()))
```

Comment:
Perhaps we should use a timestamp passed the deleteRetentionMs (e.g., time.milliseconds() - logConfig.deleteRetentionMs  -1) and make sure the tombstone is still preserved (since before the tombstone is cleaned at least once, it shouldn't be deleted) after cleaning?